apps/bfd-pipeline/bfd-pipeline-rif-extract/src/test/java/gov/cms/bfd/pipeline/rif/extract/synthetic/SyntheticDataFixer4.java

```diff
@@ -0,0 +1,525 @@
+package gov.cms.bfd.pipeline.rif.extract.synthetic;
+
+import gov.cms.bfd.model.rif.BeneficiaryColumn;
+import gov.cms.bfd.model.rif.CarrierClaimColumn;
+import gov.cms.bfd.model.rif.InpatientClaimColumn;
+import gov.cms.bfd.model.rif.PartDEventColumn;
+import gov.cms.bfd.model.rif.RifFileType;
+import gov.cms.bfd.model.rif.parse.RifParsingUtils;
+import gov.cms.bfd.pipeline.rif.extract.LocalRifFile;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVPrinter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A one-off utility application for modifying the fixed-with-negative-ids sample data that was
+ * produced on 2017-11-27 and then fixed by {@link SyntheticDataFixer}. This class adds additional
+ * beneficiary columns from the BSF file.
+ */
+public final class SyntheticDataFixer4 {
+  private static class MBIPool {
+    private int i = 0;
+
+    private String template = "%dS%s%d%s%s%d%s%s%d%d";
+    private String[] alpha = {
+      "A", "C", "D", "E", "F", "G", "H", "J", "K", "M", "N", "P", "Q", "R", "S", "U", "V", "W", "X",
+      "Y"
+    };
+    private String[] alphanumeric = {
+      "0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "A", "C", "D", "E", "F", "G", "H", "J", "K",
+      "M", "N", "P", "Q", "R", "S", "U", "V", "W", "X", "Y"
+    };
+
+    public String next() {
```

Comment:
The goal of this method is to always generate the same MBI for a given `i`, correct? If so, that appears to be correct.

Reply:
That's correct